src/python_testing/mdns_discovery/mdns_discovery.py

```diff
@@ -78,6 +82,25 @@ class MdnsServiceType(Enum):
     BORDER_ROUTER = "_meshcop._udp.local."
 
 
+class MdnsServiceListener(ServiceListener):
+    """
+    A service listener required for the TXT record data to get populated and come back
+    """
+
+    def __init__(self):
+        self.updated_event = asyncio.Event()
+
+    def add_service(self, zeroconf: Zeroconf, service_type: str, name: str) -> None:
+        sleep(0.25)
```

Comment:
I've tested this with and without a small delay, for some reason without it the TXT record won't populate sometimes. I wasn't able to dig to deep enough to know why this is.
I'll try it without it again since there's been updates to zeroconf.

Reply:
Still intermittently fails, I can do half of the delay and it still works though.

Reply:
Updated to half of original delay.